File: 2023/2/part1.py
from sys import stdin 

MAX_COUNT = {"red": 12, "green": 13, "blue":14}

def main():
    total = 0
    for line in stdin: 
        line = line.split(":")
        game_id = int(line[0].split()[1])
        games = line[1].strip().split(";")

        possible = True
        for g in games: 
            colours = g.split(",")
            for c in colours:
                c = c.strip().split()
                colour = c[1]
                num = int(c[0])
                if num > MAX_COUNT[colour]:
                    possible = False

        if possible: 
            total += game_id
            
    print(total)

if __name__=='__main__': 
    main()

File: 2023/2/kotlin/part1.kt
fun main() { 
    val MAX = mapOf(
        "red" to 12,
        "green" to 13,
        "blue" to 14
    )

    var total = 0

    val games = generateSequence(::readLine)
        .map { it.split(":") }
        .toList()
    
    games.forEach { game -> 
        val gameNum = game[0].split(" ")[1].toInt()
        val values = game[1].split(",",";")
        val impossible = values.map { it.split(" ") }
            .filter { MAX.getValue(it[2]) < it[1].toInt() }
        if (impossible.isEmpty()) { 
            total = total + gameNum
        }
    }
    println(total)
   
}
